lib/rtp_vp9/vp9_payload_descriptor.ex

```diff
@@ -0,0 +1,347 @@
+  end
+
+  defmodule SSDimension do
+    @type t :: %__MODULE__{
+            width: 0..65535,
+            height: 0..65535
+          }
+
+    @enforce_keys [:width, :height]
+    defstruct @enforce_keys
+  end
+
+  defmodule ScalabilityStructure do
+    alias Membrane.RTP.VP9.PayloadDescriptor
+    alias Membrane.RTP.VP9.PayloadDescriptor.{SSDimension, PGDescription}
+
+    @type t :: %__MODULE__{
+            first_octet: PayloadDescriptor.first_octet(),
+            dimensions: list(SSDimension.t()),
+            pg_descriptions: list(PGDescription.t())
+          }
+
+    @enforce_keys [:first_octet]
+    defstruct [:first_octet, dimensions: [], pg_descriptions: []]
+  end
+
+  @type t :: %__MODULE__{
+          first_octet: first_octet(),
+          picture_id: picture_id(),
+          tid: tid(),
+          u: u(),
+          sid: sid(),
+          d: d(),
+          p_diffs: list(p_diff()),
+          tl0picidx: tl0picidx(),
+          scalability_structure: ScalabilityStructure.t()
+        }
+
+  defstruct [
+    :first_octet,
+    :picture_id,
+    :tid,
+    :u,
+    :sid,
+    :d,
+    :tl0picidx,
+    :scalability_structure,
+    p_diffs: []
+  ]
+
+  @spec parse_payload_descriptor(binary()) :: {:error, :malformed_data} | {:ok, {t(), binary()}}
+  def parse_payload_descriptor(raw_payload)
+
+  def parse_payload_descriptor(<<header::binary-size(1), rest::binary()>>)
+      when byte_size(rest) > 0 do
+    <<i::1, _p::1, _l::1, f::1, _bevz::4>> = header
+
+    if i != f do
+      {:error, :malformed_data}
+    else
+      case get_pid(header, rest, %__MODULE__{first_octet: :binary.decode_unsigned(header)}) do
+        {:ok, {descriptor_acc, rest}} ->
+          case get_layer_indices(header, rest, descriptor_acc) do
+            {:ok, {descriptor_acc, rest}} ->
+              case get_pdiffs(header, rest, 0, descriptor_acc) do
+                {:ok, {descriptor_acc, rest}} ->
+                  case get_scalability_structure(header, rest) do
+                    {:ok, {ss, rest}} ->
+                      {:ok, {%{descriptor_acc | scalability_structure: ss}, rest}}
+
+                    _ ->
+                      {:error, :malformed_data}
+                  end
+
+                _ ->
+                  {:error, :malformed_data}
+              end
+
+            _ ->
+              {:error, :malformed_data}
+          end
+
+        _ ->
+          {:error, :malformed_data}
+      end
+    end
+  end
+
+  def parse_payload_descriptor(_), do: {:error, :malformed_data}
+
+  # no picture id (PID)
+  defp get_pid(<<0::1, _::7>>, rest, descriptor_acc) when byte_size(rest) > 0,
+    do: {:ok, {descriptor_acc, rest}}
+
+  # picture id (PID) present
+  defp get_pid(<<1::1, _::7>>, <<pid::binary-size(1), rest::binary()>>, descriptor_acc)
+       when byte_size(rest) > 0 do
+    case pid do
+      <<0::1, _rest_of_pid::7>> ->
+        {:ok, {%{descriptor_acc | picture_id: :binary.decode_unsigned(pid)}, rest}}
```

Comment:
```suggestion
  defp get_pid(<<1::1, _::7>>, <<pid, rest::binary()>>, descriptor_acc)
       when byte_size(rest) > 0 do
    case pid do
      <<0::1, _rest_of_pid::7>> ->
        {:ok, {%{descriptor_acc | picture_id: pid}, rest}}
```

Applies to each use of `decode_unsigned`